oscarlab/graphene
# Improve GCC regression tests and deprecate the "gcc_test_files" submodule

This PR cleans up the GCC-related test files and rewrites the regression tests using our own framework.

Details of the PR:
* Remove Makefile rules for building GCC locally (no longer necessary)
* Remove tarball files that contain source code for building GCC
* Deprecate the gcc_test_files submodule; most of the sample files can be downloaded from https://people.csail.mit.edu/smcc/projects/single-file-programs/, except `helloworld.c`; one less repository to maintain
* Add Makefile rules for downloading the test files and verifying the checksum
* One sample file (`gzip.c`) is buggy to compile with GCC 5.5; a patch is added to fix the bug
* Move regression code from Makefile to two python scripts (`test.py` and `test-huge.py`), using our regression framework
* Add compilation of single-file GCC code as a regression test

How to test:
```
make
cd LibOS/shim/test/apps/gcc
make regression
```

Expected output:
```
--2018-11-29 23:05:52--  https://people.csail.mit.edu/smcc/projects/single-file-programs/gzip.c
Resolving people.csail.mit.edu (people.csail.mit.edu)... 128.30.2.133
Connecting to people.csail.mit.edu (people.csail.mit.edu)|128.30.2.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 272247 (266K) [text/x-csrc]
Saving to: ‘gzip.c’

gzip.c                                                    100%[====================================================================================================================================>] 265.87K   766KB/s    in 0.3s

2018-11-29 23:05:53 (766 KB/s) - ‘gzip.c’ saved [272247/272247]

--2018-11-29 23:05:53--  https://people.csail.mit.edu/smcc/projects/single-file-programs/oggenc.c
Resolving people.csail.mit.edu (people.csail.mit.edu)... 128.30.2.133
Connecting to people.csail.mit.edu (people.csail.mit.edu)|128.30.2.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1717656 (1.6M) [text/x-csrc]
Saving to: ‘oggenc.c’

oggenc.c                                                  100%[====================================================================================================================================>]   1.64M  1.45MB/s    in 1.1s

2018-11-29 23:05:55 (1.45 MB/s) - ‘oggenc.c’ saved [1717656/1717656]

--2018-11-29 23:05:55--  https://people.csail.mit.edu/smcc/projects/single-file-programs/gcc.c.bz2
Resolving people.csail.mit.edu (people.csail.mit.edu)... 128.30.2.133
Connecting to people.csail.mit.edu (people.csail.mit.edu)|128.30.2.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 3288025 (3.1M) [application/x-bzip2]
Saving to: ‘gcc.c.bz2’

gcc.c.bz2                                                 100%[====================================================================================================================================>]   3.13M  1.81MB/s    in 1.7s

2018-11-29 23:05:57 (1.81 MB/s) - ‘gcc.c.bz2’ saved [3288025/3288025]

sha256sum --check test_files_checksums
test_files/bzip2.c: OK
test_files/gcc.c.bz2: OK
test_files/gzip.c: OK
test_files/oggenc.c: OK

Building manifest...


Testing...
[Success] helloworld.c
[Success] bzip2.c
[Success] gzip-patched.c


Building manifest with HUGE=1...


Testing...
[Success] gcc.c
```



---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/oscarlab/graphene/335)
